google/cloud/spanner_dbapi/cursor.py

```diff
@@ -175,18 +175,40 @@ def execute(self, sql, args=None):
                 sql, params = sql_pyformat_args_to_spanner(sql, args)
 
                 statement = Statement(
-                    sql, params, get_param_types(params), ResultsChecksum(),
-                )
-                (self._result_set, self._checksum,) = self.connection.run_statement(
-                    statement
+                    sql,
+                    params,
+                    get_param_types(params),
+                    ResultsChecksum(),
                 )
+                (
+                    self._result_set,
+                    self._checksum,
+                ) = self.connection.run_statement(statement)
                 self._itr = PeekIterator(self._result_set)
                 return
 
             if classification == parse_utils.STMT_NON_UPDATING:
                 self._handle_DQL(sql, args or None)
             elif classification == parse_utils.STMT_INSERT:
-                _helpers.handle_insert(self.connection, sql, args or None)
+
+                # Read INFORMATION_SCHEMA.COLUMNS to get the Spanner types for
```

Comment:
I would strongly recommend that you only do this as a workaround when running against the emulator, as real Spanner seems to accept the parameter type to be missing. Otherwise you will be slowing down statements unnecessarily in production environments.
Furthermore, you probably need to do this for all types of statements on the emulator, not only insert statements.